API breakage in 2.5.1

Despite
```
No API/ABI break compared to v2.5.0
```
in the 2.5.1 release notes, an upgrade to 2.5.1 in pkgsrc broke some other software.

Two examples:
[Pillow](https://us-central.manta.mnx.io/pkgsrc/public/reports/upstream-trunk/20240226.2249/py311-Pillow-10.2.0/build.log)
```
src/libImaging/Jpeg2KDecode.c: In function 'j2k_decode_entry':
src/libImaging/Jpeg2KDecode.c:673:5: error: too few arguments to function 'opj_stream_set_user_data'
  673 |     opj_stream_set_user_data(stream, state);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from src/libImaging/Jpeg2K.h:11,
                 from src/libImaging/Jpeg2KDecode.c:21:
/home/pbulk/build/graphics/py-Pillow/work/.buildlink/include/openjpeg-2.5/openjpeg.h:1235:27: note: declared here
 1235 | OPJ_API void OPJ_CALLCONV opj_stream_set_user_data(opj_stream_t* p_stream,
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~
```

[ghostscript](https://us-central.manta.mnx.io/pkgsrc/public/reports/upstream-trunk/20240226.2249/ghostscript-agpl-10.02.1nb2/build.log)
```
./base/sjpx_openjpeg.c: In function 's_opjd_process':
./base/sjpx_openjpeg.c:760:5: warning: "OPJ_VERSION_MAJOR" is not defined, evaluates to 0 [-Wundef]
  760 | #if OPJ_VERSION_MAJOR >= 2 && OPJ_VERSION_MINOR >= 1
      |     ^~~~~~~~~~~~~~~~~
./base/sjpx_openjpeg.c:763:13: error: too few arguments to function 'opj_stream_set_user_data'
  763 |             opj_stream_set_user_data(state->stream, &(state->sb));
      |             ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from ./base/sjpx_openjpeg.h:25,
                 from ./base/sjpx_openjpeg.c:24:
/home/pbulk/build/print/ghostscript-agpl/work/.buildlink/include/openjpeg-2.5/openjpeg.h:1235:27: note: declared here
 1235 | OPJ_API void OPJ_CALLCONV opj_stream_set_user_data(opj_stream_t* p_stream,
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~
```

I guess the software used some now invalid way to check for openjpeg versions, e.g. [Pillow](https://github.com/python-pillow/Pillow/blob/0fc9b9183fa8a97f37e0c06e64505e3813ce7c7d/src/libImaging/Jpeg2KDecode.c#L672)
```
/* OpenJPEG 2.0 doesn't have OPJ_VERSION_MAJOR */
#ifndef OPJ_VERSION_MAJOR
    opj_stream_set_user_data(stream, state);
#else
     opj_stream_set_user_data(stream, state, NULL);
...
```

--- a/src/lib/openjp2/openjpeg.h
+++ b/src/lib/openjp2/openjpeg.h
@@ -138,6 +138,8 @@ typedef int64_t  OPJ_OFF_T; /* 64-bit file offset type */
 #include <stdio.h>
 typedef size_t   OPJ_SIZE_T;
 
+#include "opj_config.h"
+
 /* Avoid compile-time warning because parameter is not used */
 #define OPJ_ARG_NOT_USED(x) (void)(x)
 

--- a/src/lib/openjp2/opj_config.h.cmake.in
+++ b/src/lib/openjp2/opj_config.h.cmake.in
@@ -1,3 +1,6 @@
+#ifndef OPJ_CONFIG_H_INCLUDED
+#define OPJ_CONFIG_H_INCLUDED
+
 /* create opj_config.h for CMake */
 
 /*--------------------------------------------------------------------------*/
@@ -7,3 +10,5 @@
 #define OPJ_VERSION_MAJOR @OPENJPEG_VERSION_MAJOR@
 #define OPJ_VERSION_MINOR @OPENJPEG_VERSION_MINOR@
 #define OPJ_VERSION_BUILD @OPENJPEG_VERSION_BUILD@
+
+#endif

